app/src/main/java/de/triology/universeadm/user/UserResource.java

```diff
@@ -26,162 +26,153 @@
  */
 
 
-
 package de.triology.universeadm.user;
 
 //~--- non-JDK imports --------------------------------------------------------
 
 import com.google.inject.Inject;
-
 import de.triology.universeadm.AbstractManagerResource;
+import de.triology.universeadm.csvimport.CSVImportManager;
 import de.triology.universeadm.group.Group;
 import de.triology.universeadm.group.GroupManager;
 
-//~--- JDK imports ------------------------------------------------------------
-
-import javax.ws.rs.DELETE;
-import javax.ws.rs.POST;
-import javax.ws.rs.Path;
-import javax.ws.rs.PathParam;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
 
 /**
  * TODO remove package cycle with group.
  *
  * @author Sebastian Sdorra <sebastian.sdorra@triology.de>
  */
 @Path("users")
-public class UserResource extends AbstractManagerResource<User>
-{
-
-  /**
-   * Constructs ...
-   *
-   *
-   * @param userManager
-   * @param groupManager
-   */
-  @Inject
-  public UserResource(UserManager userManager, GroupManager groupManager)
-  {
-    super(userManager);
-    this.userManager = userManager;
-    this.groupManager = groupManager;
-  }
-
-  //~--- methods --------------------------------------------------------------
-
-  /**
-   * Method description
-   *
-   *
-   * @param username
-   * @param groupname
-   *
-   * @return
-   */
-  @POST
-  @Path("{user}/groups/{group}")
-  public Response addMembership(@PathParam("user") String username,
-    @PathParam("group") String groupname)
-  {
-    Response.ResponseBuilder builder;
-
-    User user = userManager.get(username);
-    Group group = groupManager.get(groupname);
-
-    if (user == null)
-    {
-      builder = Response.status(Response.Status.NOT_FOUND);
+public class UserResource extends AbstractManagerResource<User> {
+    /**
+     * Constructs ...
+     *
+     * @param userManager
+     * @param groupManager
+     */
+    @Inject
+    public UserResource(UserManager userManager, GroupManager groupManager, CSVImportManager csvImportManager) {
+        super(userManager);
+        this.userManager = userManager;
+        this.groupManager = groupManager;
+        this.csvImportManager = csvImportManager;
     }
-    else if (group == null)
-    {
-      builder = Response.status(Response.Status.BAD_REQUEST);
-    }
-    else if (user.getMemberOf().contains(groupname))
-    {
-      builder = Response.status(Response.Status.CONFLICT);
+
+    //~--- methods --------------------------------------------------------------
+    @POST
+    @Path("/import")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response importUsers(InputStream inputStream) {
+        Response.ResponseBuilder builder;
+        try {
+            this.csvImportManager.importUsers(inputStream);
+            builder = Response.status(Response.Status.OK);
+        } catch (IOException | IllegalArgumentException e) {
```

Comment:
An IOException should not lead to a response with status code 400. It's an application error (500) if this exception ocurres.